### PR TITLE
Fix Deposit Details reactivity

### PR DESCRIPTION
**Pull request by kpyszkowski**
There is an issue with data returned by the `useFetchDepositDetails` hook. It is not reactive, once read it's not updated as the process proceeds. This PR resolves this issue.

### CODE REVIEW

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/DepositDetails.tsx:81]: Why are we removing this piece of code here? Without that, the `txConfirmations` in the store won't be updated.

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/DepositDetails.tsx:85]: Besides that, removing this from the hook break the `Minting Timeline`. 

I'm wondering if putting `updateState("depositDetailsStep", "bitcoin-confirmations")` in `return` was a good idea, because it looks it doesn't work as expected tbh 🤔 But still, it would be nice to clear the `depositDetailsStep` somehow when we leave the page.

File: src/hooks/tbtc/useFetchDepositDetails.ts
Comment (michalsmiarowski) [inline src/hooks/tbtc/useFetchDepositDetails.ts:136]: I would prefer to not put it here and update like that, because it might lead to unexpected behaviour. For example we can have 5/6 confirmations for the given deposit. Let's say we go to the deposit details page and the fetch starts. We are at this line:
```
const confirmations =
          (await threshold.tbtc.getTransactionConfirmations(btcTxHash)) ??
          txConfirmations
```
and `confirmations` is equal to 5. After that (but before fetch is over) the 6th confirmation happens so we update the `confirmations` in the state with the value of 6. Then, at the end of the fetch, we overwrite the value with `5` which is out-dated by now.

Same can happen with tx hashes.

The safer way would be to move those two calls:
```
const { optimisticMintingFinalizedTxHash, optimisticMintingRequestedTxHash } =
    useSubscribeToOptimisticMintingEvents(depositKey)
  const { txConfirmations } = useTbtcState()
```

to `DepositDetails.tsx` and the use double question mark to choose the right value. The values from the state should have a priority.

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some comments to look at before the merge

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/DepositDetails.tsx]: Ref commit: 6e793eb3

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/DepositDetails.tsx]: Reverted, ref commit: 5d92563d

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/DepositDetails.tsx]: > Besides that, removing this from the hook break the Minting Timeline.

Please elaborate on that, I don't understand what is not working.

> I'm wondering if putting [...] in `return` was a good idea 
> [...] it looks it doesn't work as expected.

That's what `useEffect`'s cleanup function is for. It's called as the element - here `DepositDetails` component is removed from the DOM and it works as expected. The value is set to the first step following the pattern presented in the `mintingStep` value behavior.

File: src/hooks/tbtc/useFetchDepositDetails.ts
Comment (kpyszkowski) [reply src/hooks/tbtc/useFetchDepositDetails.ts:136]: As long as the value is updated, the change is captured by `useEffect` hook and the `depositData` state is updated accordingly. Notice I'm aggregating the update upon the most recent state value. 
Actually the `isFetching` flag is there only to provide feedback for loading state, nothing more.
Spreading correlated state update across multiple files/hooks is confusing, difficult to maintain and should be avoided. That's why I'm trying to keep everything in one place. If you call `useFetchDepositData` you expect to get **complete** deposit data. How could I know to get confirmations count from another place and transaction hashes from one another place? It's overcomplicated - especially if the shape of data provided by the `useFetchDepositData` hook indicates both confirmations count and hashes are there and (intuitively) should be there up to date but isn't.

If you disagree and you have idea how to resolve it, please feel free to commit your changes. I ain't got a what to do with it 😆

File: src/pages/tBTC/Bridge/DepositDetails.tsx
Comment (michalsmiarowski) [reply src/pages/tBTC/Bridge/DepositDetails.tsx:85]: > Please elaborate on that, I don't understand what is not working.

Yeah, I didn't describe what was wrong, my bad. 

It was not working when the if statement and dependency array was removed (you've already reverted the changes [here](https://github.com/threshold-network/bitcoin-on-base/pull/45#discussion_r1521340041). The reason for that is because without the dependency array, the cleanup function was fired every time the `DepositDetails` component was re-rednered, and it is actually re-rendered a lot of times when some values changes. Because of that, the timeline component always displayed the first step, because the cleanup function was fired when it shouldn't be. Here is a quick video demonstrating that:

https://github.com/threshold-network/bitcoin-on-base/assets/40306834/2464dcd2-b336-4f3d-bf04-8e89c8bd4d56

Both deposit in the viedeo are completed. I had to refresh my memory about useEffect's cleanup function to make figure out why it didn't work properly here. Basically, this is a quick summary of how clean-up function behaves based on dependency array:
1. No Dependency Array: Cleanup runs after every render, before the next effect or on unmount, ensuring effects are properly cleaned up in a constantly updating component.

2. Empty Dependency Array (`[]`): Cleanup runs once, when the component unmounts, mirroring a setup that only needs to run and clean up a single time (like componentDidMount and componentWillUnmount).

3. With Dependencies (`[deps]`): Cleanup runs when any dependency changes, prior to the next effect to handle updated dependencies, and on unmount, allowing for targeted effect management and resource cleanup based on specific state or prop changes.

This means that in our current case, the cleanup function will fire again when (for example) the new confirmation for the bitcoin transaction arrives, thus set the `depositDetailsStep` to `bitcoin-confirmations` . It will work anyway, because we have another `useEffect` below with `confirmations` in dependency array that will set the proper step, but I think I came up with better solution that would be less bug-prone. I've separated the cleanup function to another useEffect with only `depositKey` as a dependency. I've also set it to `undefined` instead of `bitcoin-transactions`.

```
useEffect(() => {
    return () => {
      updateState("depositDetailsStep", undefined)
    }
  }, [depositKey])
```

Result:

https://github.com/threshold-network/bitcoin-on-base/assets/40306834/1f74cb6e-6cf2-4faf-851e-eb9ab6074550

What do you think @kpyszkowski? 

Here is the commit with the fix: c7ec0af3abeb8737823af94acf1ddc248c2658b7

Comment (kpyszkowski) [reply src/pages/tBTC/Bridge/DepositDetails.tsx:85]: Arghh, my bad. I didn't notice I removed the dependency array. Thanks. LGTM

File: src/hooks/tbtc/useFetchDepositDetails.ts
Comment (michalsmiarowski) [reply src/hooks/tbtc/useFetchDepositDetails.ts]: I've improvde it by moving the deposit details data to redux store. Please see those commits:

397810d59511727dd61b3d1aaf2f29304052feea (see this one's commit message for more info)
e0df1c7d691a1dbfc8813e6e1b9b6ad269ca449c
94fc793c5b63f51b5011c8e3636967e84da2c719

**Review by michalsmiarowski — APPROVED** [review]
LGTM

Would appreciate a quick look at https://github.com/threshold-network/bitcoin-on-base/pull/45#discussion_r1524205424 @kpyszkowski 😉 

EDIT:

I will merge it and if we find any errors we will address them in separate PRs

**Pull request merged**
